iModelCore/ECPresentation/Source/Shared/ValueHelpers.cpp

```diff
@@ -363,13 +360,17 @@ rapidjson::Document ValueHelpers::GetJsonFromArrayValue(IECSqlValue const& sqlVa
                 doc.PushBack(GetJsonFromStructValue(*v.GetColumnInfo().GetStructType(), v, &doc.GetAllocator()), doc.GetAllocator());
                 break;
             case ValueKind::VALUEKIND_Array:
-                {
-                Utf8String extendedArrayType = v.GetColumnInfo().GetProperty()->GetIsPrimitiveArray() ? v.GetColumnInfo().GetProperty()->GetAsPrimitiveArrayProperty()->GetExtendedTypeName() : "";
-                doc.PushBack(GetJsonFromArrayValue(v, &doc.GetAllocator(), extendedArrayType), doc.GetAllocator());
+                doc.PushBack(GetJsonFromArrayValue(v, &doc.GetAllocator()), doc.GetAllocator());
                 break;
-                }
             case ValueKind::VALUEKIND_Primitive:
-                doc.PushBack(GetJsonFromPrimitiveValue(v.GetColumnInfo().GetDataType().GetPrimitiveType(), v, &doc.GetAllocator(), extendedType), doc.GetAllocator());
+                ECPropertyCP property = v.GetColumnInfo().GetOriginProperty();

```

Comment:
Generally, if a function returns a pointer, it means it may return a `nullptr` (otherwise it would return a reference). Unless you're 100% sure that it's not going to do that, it's better to just add a null check:
```cpp
if (property == nullptr)
    DIAGNOSTICS_HANDLE_FAILURE(DiagnosticsCategory::Default, "<error message here, ideally with some information about the column>");
```

Reply:
fixed?